asmcomp/selectgen.ml

```diff
@@ -42,6 +42,10 @@ let oper_result_type = function
   | Cintoffloat -> typ_int
   | Craise _ -> typ_void
   | Ccheckbound _ -> typ_void
+  | Cperform -> typ_addr
```

Comment:
`typ_addr` is correct here, but will need changing when we merge to current trunk.

Multicore is currently based on the 4.02.1 code. 4.02.1 has the following `machtype_components`:
 - `Addr`: general values, integers or pointers
 - `Int`: tagged integers (subtype of `Addr`)
 - `Float`: unboxed floats

A `machtype` is an array of the above, so `typ_addr = [| Addr |]`, etc. `typ_void` is `[| |]`, for operations that produce no result, not even unit.

It transpired that `Addr` was used for intermediate values, like derived pointers into the middle of an array, which are unsafe to expose to the GC. So, recently, `Addr` was split into:
 - `Val`: general values, integers or pointers
 - `Addr`: derived pointers which must not be live by the next safepoint.

So, `typ_addr` is correct now, but should change to `typ_val` when we merge. The naming is unfortunate.